src/key.rs

```diff
@@ -411,17 +496,51 @@ impl PublicKey {
         }
     }
 
-    /// Adds a second key to this one, returning the sum. Returns an error if
-    /// the result would be the point at infinity, i.e. we are adding this point
-    /// to its own negation
+    /// Adds a second key to this one, returning the sum.
+    ///
+    /// # Errors
+    ///
+    /// If the result would be the point at infinity, i.e. adding this point to its own negation.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// # #[cfg(feature="rand")] {
+    /// use secp256k1::{rand, Secp256k1};
+    ///
+    /// let secp = Secp256k1::new();
+    /// let mut rng = rand::thread_rng();
+    /// let (_, pk1) = secp.generate_keypair(&mut rng);
+    /// let (_, pk2) = secp.generate_keypair(&mut rng);
+    /// let sum = pk1.combine(&pk2).expect("It's improbable to fail for 2 random public keys");
+    /// # }
+    ///```
     pub fn combine(&self, other: &PublicKey) -> Result<PublicKey, Error> {
         PublicKey::combine_keys(&[self, other])
     }
 
-    /// Adds the keys in the provided slice together, returning the sum. Returns
-    /// an error if the result would be the point at infinity, i.e. we are adding
-    /// a point to its own negation, if the provided slice has no element in it,
-    /// or if the number of element it contains is greater than i32::MAX.
+    /// Adds the keys in the provided slice together, returning the sum.
```

Comment:
Perhaps this example should use three keys instead of two?

Reply:
It crossed my mind during dev, if you thought it too lets do it. Added as a separate patch.